src/utils.jl

```diff
@@ -133,3 +133,69 @@ function get_problem_data(model::MOI.AbstractOptimizer)
     
     return Q, q, G, h, A, b, nz, var_idx, nineq, ineq_con_idx, neq, eq_con_idx
 end
+
+
+"""
+    projection of vector `z` on MOI set `cone`
+"""
+function π(cone::MOI.AbstractVectorSet, z::Array{Float64})
+    if cone isa MOI.Zeros
```

Comment:
Dispatch on cone types instead of doing the conditionals

Reply:
for instance:
```julia
function π(cone::MOI.Zeros, z::Array{Float64})
...
end
function π(cone::MOI.Nonnegatives, z::Array{Float64})
...
end
```

Reply:
done